### PR TITLE
 Add stream assert methods and matchers

### PR DESCRIPTION
**Pull request by sponomarev**
Assert methods were backported from rails/rails#34740.
RSpec matchers were created from scratch.

### CODE REVIEW

File: features/channel_specs/channel_spec.feature [hunk 1]
```diff
@@ -106,10 +121,10 @@ Feature: channel spec
         stub_connection user_id: 42
         subscribe(room_id: 1)
 
-        expect(streams).to include("chat_1")
+        expect(subscription).to have_stream_from("chat_1")
 
         perform :leave
-        expect(streams).to eq []
+        expect(subscription).not_to have_stream
```
Comment (palkan) [inline features/channel_specs/channel_spec.feature]: Can we rename it to `have_streams`?)

File: spec/dummy/app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class User
+  include GlobalID::Identification
+
+  attr_reader :id
+
+  def initialize(id)
+      @id = id
```
Comment (palkan) [inline spec/dummy/app/models/user.rb]: Looks like RuboCop ignores this file))

**Comment by sponomarev** [comment]
@palkan 😄

**Comment by palkan** [comment]
tenor

**Pull request merged**